accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-ariaLabeling-rules.ts

```diff
@@ -20,8 +20,174 @@ import { ARIADefinitions } from "../../../aria/ARIADefinitions";
+                            (els[j].tagName == "FORM") ||
+                            (els[j].tagName == "HEADER") ||
+                            (els[j].tagName == "MAIN") ||
+                            (els[j].tagName == "NAV") ||
+                            (els[j].tagName == "SECTION") ||
+                            (els[j].tagName == "FORM");
+
+                        let roleNameTrigger = false;
+                        if (els[j].hasAttribute("role")) {
+                            roleNameTrigger =
+                                (els[j].getAttribute("role").includes("complementary")) ||
+                                (els[j].getAttribute("role").includes("contentinfo")) ||
+                                (els[j].getAttribute("role").includes("form")) ||
+                                (els[j].getAttribute("role").includes("banner")) ||
+                                (els[j].getAttribute("role").includes("main")) ||
+                                (els[j].getAttribute("role").includes("navigation")) ||
+                                (els[j].getAttribute("role").includes("region")) ||
+                                (els[j].getAttribute("role").includes("form")) ||
+                                (els[j].getAttribute("role").includes("search"));
+                        }
+                        if (tagNameTrigger || roleNameTrigger) {
+                            // Nearest parent-landmark found
+                            navigationNodesParents.push(els[j])
+                            break
+                        }
+                        if (j == els.length - 1) { // This node is at the head of the file so it does not have a parent
+                            navigationNodesParents.push(null) // TODO might want to change to NULL
+                            break
+                        }
+                    }
+                }
+
+                let navigationNodesComputedLabels = [];
+                for (let i = 0; i < navigationNodes.length; i++) { // Loop over all the landmark nodes
+                    navigationNodesComputedLabels.push(ARIAMapper.computeName(navigationNodes[i]))
+                }
+                for (let i = 0; i < navigationNodesParents.length; i++) { // Loop over all the parents of the landmark nodes to find duplicates
+                    let matchFound = false;
+                    let pass_0_flag = false;
+                    for (let j = 0; j < navigationNodesParents.length; j++) {
+                        if ((navigationNodesParents[i] === null) || (navigationNodesParents[j] === null)) {
+                            // We are looking at root node
+                            continue
+                        }
+                        if ((navigationNodes[i] === null) || (navigationNodes[j] === null)) {
+                            // Strange case... should not happen
+                            continue
+                        }
+                        if (j == i) {
```

Comment:
This is where the index construct can be better, but please use ===. `of` may still work with navNode1 === navNode2, but I'm not really sure. Safer to keep indexes.

Reply:
Makes sense. No code change needed so resolving.